File: jiyong/Python/PGS_72413/PGS_72413.py
def solution(n, s, a, b, fares):
    INF = 2147483647
    dist_table = [[INF] * (n + 1) for _ in range(n + 1)]
    for c, d, f in fares:
        dist_table[c][c] = 0
        dist_table[d][d] = 0
        dist_table[c][d] = f
        dist_table[d][c] = f

    for k in range(1, n + 1):
        for r in range(1, n + 1):
            for c in range(1, n + 1):
                dist_table[r][c] = min(dist_table[r][k] + dist_table[k][c], dist_table[r][c])
    print(*dist_table, sep='\n')
    return min([dist_table[x][a] + dist_table[x][b] + dist_table[x][s] for x in range(1, n + 1)])


print(solution(5, 4, 5, 1, [[1, 2, 1], [1, 3, 1], [1, 4, 5], [2, 3, 2], [3, 4, 2], [2, 5, 1], [3, 5, 2], [4, 5, 5]]))

File: yerim/yerimRepo/src/week7/PGS72413/PGS72413.java
package week7.PGS72413;

import java.util.*;
// 갱신 -> Math.min(합승x, 합승 o)
class Solution{
    static int[][] map;
    static int[] d; // 거리테이블 사이즈 ....
    public int solution(int n, int s, int a, int b, int[][] fares) {
        map = new int[n+1][n+1];

        // map의 그래프 값 초기화
        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= n; j++) {
                // 매 지점에서 낼 수 있는 최대 택시 요금으로
                map[i][j] = 200*100000+1;

            }
            map[i][i] = 0;

        }
        for (int i = 0; i < fares.length; i++) {
            int a1 = fares[i][0];
            int b1 = fares[i][1];
            int c1 = fares[i][2];

            map[a1][b1] = c1;
            map[b1][a1] = c1;

        }

        // 플로이드-워셜 알고리즘
        for (int k = 1; k <= n; k++) {
            for (int i = 1; i <= n; i++) {
                for (int j = 1; j <= n; j++) {
                    map[i][j] = Math.min(map[i][j], map[i][k]+map[k][j]);
                }
            }
        }

        // 최저 택시요금
        int minRate = Integer.MAX_VALUE;
        for (int i = 1; i <= n; i++) {
            minRate = Math.min(minRate, map[s][i] + map[i][a] + map[i][b]);
        }
        return minRate;
    }
}